python/ray/serve/_private/deployment_state.py

```diff
@@ -1608,6 +1609,26 @@ def get_total_num_requests(self) -> float:
             RAY_SERVE_COLLECT_AUTOSCALING_METRICS_ON_HANDLE
             or len(running_replicas) == 0
         ):
+            # Drop metrics for handles that haven't sent an update in awhile.
+            # This is expected behavior for handles that were on replicas or
+            # proxies that have been shut down.
+            timeout_s = max(
+                2 * self.autoscaling_policy_manager.get_metrics_interval_s(),
+                RAY_SERVE_MIN_HANDLE_METRICS_TIMEOUT_S,
+            )
+            for handle_id, handle_metric in list(self.handle_requests.items()):
+                if time.time() - handle_metric.timestamp >= timeout_s:
+                    del self.handle_requests[handle_id]
+                    if (
+                        handle_metric.queued_requests
+                        + sum(handle_metric.running_requests.values())
+                        > 0
+                    ):
+                        logger.info(
+                            f"Dropping stale metrics for handle '{handle_id}' "
+                            f"(no updates received for {timeout_s:.1f}s)."
+                        )
```

Comment:
Can you log the number of queued requests at the handle here? If this causes odd autoscaling behavior, that would help us figure out how much of an impact this staleness-handling has.

Reply:
done